modules/lido/ethereum-validators-monitoring/default.nix

```diff
@@ -27,34 +27,58 @@
 
       db = {
         db-host = mkOption {
-          type = types.nullOr types.str;
+          type = types.str;
           description = "Clickhouse server host.";
         };
 
         db-user = mkOption {
-          type = types.nullOr types.str;
+          type = types.str;
           description = "Clickhouse server user.";
         };
 
-        db-password = mkOption {
-          type = types.nullOr types.str;
+        db-password-file = mkOption {
+          type = types.path;
           description = "Clickhouse server password.";
         };
 
         db-name = mkOption {
-          type = types.nullOr types.str;
+          type = types.str;
           description = "Clickhouse server DB name.";
         };
 
         db-port = mkOption {
-          type = types.nullOr types.port;
+          type = types.port;
           default = 8123;
           description = "Clickhouse server port.";
         };
       };
     };
 
     config = lib.mkIf (eachService != {}) {
+      systemd.services.ethereum-validators-monitoring-preStart = {
+        after = ["network.target"];
+        wantedBy = ["multi-user.target"];
+        script = ''
+          mkdir -p /var/lib/ethereum-validators-monitoring
+          mkdir -p /var/lib/clickhouse
+
+          # Create file if it doesn't exist, truncate it if does
+          touch /var/lib/ethereum-validators-monitoring/env
+          echo "" > /var/lib/ethereum-validators-monitoring/env
+
+          touch /var/lib/clickhouse/env
+          echo "" > /var/lib/clickhouse/env
+
+          chmod 600 /var/lib/ethereum-validators-monitoring/env
+          chmod 600 /var/lib/clickhouse/env
+
+
+          echo DB_PASSWORD="$(cat ${db.db-password-file})" >> /var/lib/ethereum-validators-monitoring/env
+          echo CLICKHOUSE_PASSWORD="$(cat ${db.db-password-file})" >> /var/lib/clickhouse/env
```

Comment:
I don't quite understand the motivation here.
The password will still exist in plain text within the `/var/lib/clickhouse/env` file.

Reply:
This service inserts the password into the appropriate vatiables after retrieving it from the secret.  This is the simplest method I've found (given that only root can read these files), as there is no way to incorporate this step into its service because we haven't bundled ethereum-validators-monitoring yet.  The systemd units of this service are not accessible through the present use of Docker.  I can place this file in a systemd unit that the package can access when we create the package.